core/domain/translation_domain.py

```diff
@@ -181,8 +181,9 @@ def add_translatable_field(
             content_value: ContentValueType. Value of the content which
                 is translatable.
         """
-        # The duplicate check is avoided here as object validation check ensures
-        # all the content_ids are unique.
+        if content_id in self.content_id_to_translatable_content:
+            raise Exception('Already added as a translatable content.')
```

Comment:
Have your error be more descriptive: "content_id %s already exists in the TranslatableContentsCollection." (i.e. include the content id for debugging purposes)

Reply:
Updated.